app/assets/javascripts/users.js.erb

```diff
@@ -264,6 +301,21 @@ function newMatchModalEvents() {
   $('.new-match-modal .skip-link').click(function() {
     $('.new-match-modal').modal('hide');
   });
+  $('.new-match-modal .modal-button').click(function() {
+    matchId = $('.new-match-user h4').data('match-id');
+    $('.new-match-modal').modal('hide');
+    $('[data-match-id="' + matchId + '"]').trigger('click');
```

Comment:
you can use just click instead `$('[data-match-id="' + matchId + '"]').click()`

`.click()` is a shortcut for `.trigger( "click" )`

https://api.jquery.com/click/#click